src/app/features/dashboard-widgets/birthday-dashboard-widget/birthday-dashboard-settings.component/birthday-dashboard-settings.component.ts

```diff
@@ -70,41 +84,111 @@ export class BirthdayDashboardSettingsComponent implements OnInit {
       },
     );
 
+    // Build availablePropertiesMap for each entity type
+    for (const ctor of this.availableEntityTypes) {
+      const schema = getEntitySchema(ctor);
+      this.availablePropertiesMap[ctor.ENTITY_TYPE] = Array.from(
+        schema.entries(),
+      )
+        .filter(([_, field]: any) => field.dataType === "date-with-age")
+        .map(([key]) => key);
+    }
```

Comment:
Following a "clean code" concept, you could move this into a private sub-method called `buildAvailablePropertiesMap()` which makes the init method shorter and replaces the comment (because the explanation is in the method name already)